### PR TITLE
Rename CI tests to minimal/extended tests

### PR DESCRIPTION
**Pull request by qinsoon**
This PR refactors the CI workflows for https://github.com/mmtk/mmtk-core/issues/935, and run some OpenJDK tests as minimal tests.

### CODE REVIEW

**Comment by qinsoon** [comment]
binding-refs
OPENJDK_BINDING_REPO=qinsoon/mmtk-openjdk
OPENJDK_BINDING_REF=mmtk-core-test

File: docs/team/ci.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+## Testing
+
+MMTK core runs CI tests *before* a pull request is merged.
+
+MMTk core sets up two sets of tests, the *minimal tests* and the *extended tests*.
+* Minimal tests run unconditionally for every pull request and for every commit. This set of tests aims to finish within half an hour.
+  This include tests from the `mmtk-core` repo and integration tests from binding repos. Integration tests with a binding in the minimal tests should
+  focus on testing MMTk core features that is exclusively used for the binding.
+* Extended tests only run for a pull request if the pull request is tagged with the label `PR-extended-testing`. This set of tests
```
Comment (wks) [inline docs/team/ci.md:11]: But `extended-tests-bindings.yml` still mentions `PR-testing`.  Should that be changed, too?

Comment (qinsoon) [reply docs/team/ci.md:11]: Yes. I made that change.

**Review by wks — APPROVED** [review]
LGTM

**Comment by qinsoon** [comment]
@wks Ruby tests kept failing in all the attempts. You may want to look into it. It should not be related to this PR, and I may merge this PR along with the OpenJDK PR.

**Comment by wks** [comment]
I can reproduce the Rust bug locally, but not stably.  It is very random.  It happens during GC when marking lines.

**Pull request merged**